File: easy/192_compare_points.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, const char * argv[]) {
    FILE *file = fopen(argv[1], "r");
    char line[1024];
    int O, P, Q, R;
    int dx, dy;

    while (fgets(line, 1024, file)) {
        sscanf(line, "%d %d %d %d", &O, &P, &Q, &R);
        dx = Q - O;
        dy = R - P;

        if (dy > 0)
            printf("N");
        else if (dy < 0)
            printf("S");

        if (dx > 0)
            printf("E");
        else if (dx < 0)
            printf("W");

        if (dx == 0 && dy == 0)
            printf("here");

        printf("\n");
    }

    return 0;
}

File: easy/192_compare_points.py
import sys

for line in open(sys.argv[1]):
    O, P, Q, R = [int(x) for x in line.split()]
    dx, dy = (Q - O, R - P)
    direction = []

    if dy > 0:
        direction.append('N')
    elif dy < 0:
        direction.append('S')

    if dx > 0:
        direction.append('E')
    elif dx < 0:
        direction.append('W')

    if len(direction):
        print(''.join(direction))
    else:
        print('here')

# above is slightly faster according to codeeval
#from __future__ import print_function
#for line in open(sys.argv[1]):
#    O, P, Q, R = [int(x) for x in line.split()]
#    dx, dy = (Q - O, R - P)
#
#    if dy > 0:
#        print('N', end='')
#    elif dy < 0:
#        print('S', end='')
#
#    if dx > 0:
#        print('E', end='')
#    elif dx < 0:
#        print('W', end='')
#
#    if dx == dy == 0:
#        print('here', end='')
#
#    print('')
